nmt_chainer/evaluation.py

```diff
@@ -184,13 +186,30 @@ def beam_search_translate(encdec, eos_idx, src_data, beam_width = 20, beam_pruni
                 tr, sc, attn = translations[num_t]
                 rescored_translations.append((tr, sc + reverse_scores[num_t], attn))
             translations = rescored_translations
-                    
-        if post_score_length_normalization == 'simple':
-            translations.sort(key = lambda x:x[1]/(len(x[0])+1), reverse = True)
-        elif post_score_length_normalization == 'google':
-            translations.sort(key = lambda x:x[1]/( pow((len(x[0])+5), length_normalization_strength) / pow(6, length_normalization_strength) ), reverse = True)
-        else:
-            translations.sort(key = operator.itemgetter(1), reverse = True)
+
+        def ranking_criterion(x):
+            if post_score_length_normalization == 'none' and post_score_coverage_penalty == 'none':
+                return operator.itemgetter(1)
+
+            length_normalization = 1
+            if post_score_length_normalization == 'simple':
+                length_normalization = len(x[0])+1
+            elif post_score_length_normalization == 'google':
+                length_normalization = pow((len(x[0])+5), length_normalization_strength) / pow(6, length_normalization_strength)
+
+            coverage_penalty = 0
+            if post_score_coverage_penalty == 'google':
```

Comment:
Just to be safe, could you add an assertion here:
assert(len(src_data[num_ex])) == x[2][0].shape[0])